src/Symfony/Component/Console/Tests/Input/ArgvInputTest.php

```diff
@@ -337,6 +343,37 @@ public function testHasParameterOptionEdgeCasesAndLimitations()
         $this->assertFalse($input->hasParameterOption('-fh'), '->hasParameterOption() returns true if the given short option is in the raw input');
     }
 
+    public function testNoWarningOnInvalidParameterOption()
+    {
+        $input = new ArgvInput(array('cli.php', '-edev'));
+
+        ini_set('display_errors', 1);
+        ini_set('display_startup_errors', 1);
+        $original = error_reporting();
+        error_reporting(E_ALL);
+        set_error_handler([$this, 'customErrorHandler']);
```

Comment:
using a custom error handler to make the test fail in case of notice looks weird to me: the PHPUnit error handler *already* does it.

Reply:
I agree that a custom error handler is not desirable; that was [my point above](https://github.com/symfony/symfony/pull/26156#issuecomment-365635210). However, I thought phpunit was not catching php *warnings* because I made an erroneous test and got a false pass. I'll circle back and do this without the custom error handler.